### ISSUE TITLE
Behaviour of pull to refresh control and associated widget is incorrect on iOS

### ISSUE THREAD

**Issue by harveynick** [flutter/flutter#11661]
## Steps to Reproduce

Any test code which uses RefreshIndicator from the material widgets.

## Description

The widget always follows the Android behaviour. On iOS the progress indicator should act as though its part of the scroll view on drag. Then on release the scroll view should stay slightly scrolled until the indicator finished animating. The built in iOS Mail app provides an example. As does the YouTube app.

**Comment by sethladd** [comment]
thanks for the report!

**Comment by eseidelGoogle** [comment]
I think we may already have a bug asking for a CupertinoRefreshIndicator, @xster would know.

**Comment by harveynick** [comment]
Yes, and there is a CupertinoActivityIndicator in the code base. That's not quite what I'm looking for, though. I do want a material styled refresh indicator, but the current one feel incredibly alien on iOS. I believe it should behave the same as Google's existing native solutions do.

**Comment by Hixie** [comment]
@harveynick This is on our radar, but can you speak to your priority for this? Is it a blocker?

**Comment by alanrussian** [comment]
This is a launch blocker for mulligan. However, we can continue coding in the meantime.

**Comment by Hixie** [comment]
@alanrussian Thanks for the update. We will prioritise this along with other material work in the coming months; if this becomes more immediately urgent (e.g. if we haven't fixed it and you're planning on launching in a few weeks) then please let us know and we will make sure to get to this sooner.

**Comment by xster** [comment]
Note to self: there should be a haptic knock bump when reaching the trigger point.

**Comment by Hixie** [comment]
This is a top priority for the current milestone.

**Comment by xster** [comment]
Seems like there are 2 different patterns now in iOS 11. One where only the scrollable content gets wrapped with a slightly bigger scroll extent that houses the refresh control. And a new pattern where the refresh control is on top of the large title sliver which also scrolls during overscroll. 

Will implement in 2 batches. One will be an API that's a Widget wrapping a Scrollable and another will be a a parameter toggle inside the CupertinoSliverNavigationBar.

**Comment by xster** [comment]
@harveynick, can you guys test out CupertinoRefreshControl and/or share a mockup of the desired design?

**Comment by realcarlos** [comment]
Could provide a demo of CupertinoRefreshControl with fetching real api ?

and could create a pull up load-more demo with CupertinoRefreshControl? Thank you !

**Comment by xster** [comment]
Should be fairly standard usage. 

For instance, replace https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/lib/demo/cupertino/cupertino_refresh_demo.dart#L56 with something like:

```dart
final Response response = await get('https://some.api/method');
setState(() => doSomethingWith(response));
```

Also consider using [stackoverflow.com](https://stackoverflow.com) for usage questions. 

Track the pull up to load more pattern separately in #14908

**Comment by KingIdee** [comment]
Hi, is this implemented yet ?

**Comment by xster** [comment]
That's https://docs.flutter.io/flutter/cupertino/CupertinoSliverRefreshControl-class.html

https://github.com/flutter/website/issues/2290 since we forgot to add it to the catalog
